### PR TITLE
[CWS] add missing check for nil-ness of ProcessContext

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

In some cases (error cases) the process context can be nil, in this case we are currently crashing with a nil-ptr deref. This PR fixes this by checking the nill-ness of the process context.

### CODE REVIEW

File: pkg/security/rules/engine.go [hunk 1]
```diff
@@ -518,6 +518,11 @@ func (e *RuleEngine) SetRulesetLoadedCallback(cb func(es *rules.RuleSet, err *mu
 
 // HandleEvent is called by the probe when an event arrives from the kernel
 func (e *RuleEngine) HandleEvent(event *model.Event) {
+	// no process context, let's skip
+	if event.ProcessContext == nil {
+		return
+	}
+
 	// don't eval event originating from myself
 	if !e.probe.Opts.DontDiscardRuntime && event.ProcessContext.Pid == e.pid {
```
Comment (safchain) [inline pkg/security/rules/engine.go:527]: why not 
if !e.probe.Opts.DontDiscardRuntime && event.ProcessContext != nil && event.ProcessContext.Pid == e.pid

Comment (paulcacheux) [reply pkg/security/rules/engine.go:527]: we could do that yes

**Review by safchain — APPROVED** [review]

**Comment by paulcacheux** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Comment by paulcacheux** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Comment by paulcacheux** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Review by pducolin — APPROVED** [review]

**Review by spikat — APPROVED** [review]

**Pull request merged**